src/idl_parser.cpp

```diff
@@ -555,7 +556,27 @@ CheckedError Parser::ParseType(Type &type) {
         return Error(
               "vector of union types not supported (wrap in table first).");
       }
-      type = Type(BASE_TYPE_VECTOR, subtype.struct_def, subtype.enum_def);
+      if (token_ == ':') {
+        if (!IsScalar(subtype.base_type)) {
+          return Error(
+                "arrays of non-scalar types not supported.");
+        }
+        NEXT();
+        if (token_ != kTokenIntegerConstant) {
```

Comment:
so the current syntax is `field : int 4`.. I guess that's cool. I would even recommend writing it as `field : int4` in examples/docs, which has a similarity to HLSL/GLSL vector types.